generate.js

```diff
@@ -57,7 +59,21 @@ function reportErrorsAndExit(errorMessages) {
 function parseCommandInputs(args, argsByName, errorMessages, targetOutputPathList) {
     // Parse the command line arguments into key-value-pairs
     extractArgs(args, argsByName, targetOutputPathList, errorMessages);
-    // Apply defaults 
+    // Apply defaults
+    var specialization = defaultSpecialization;
+    if (argsByName.specialization) {
+        specialization = argsByName.specialization;
+    }
+    var specializationFile = path.resolve("generate-" + specialization + ".ts");
+    try {
+        specializationContent = require(specializationFile);
```

Comment:
You can require a .ts file?
NICE!
I would have instinctively required the .js file.

Reply:
Apparently it works, or it looks like it does :)

---
In reply to: [198237250](https://github.com/PlayFab/SDKGenerator/pull/142#discussion_r198237250) [](ancestors = 198237250)